src/diffusers/schedulers/scheduling_flow_match_euler_discrete.py

```diff
@@ -129,7 +129,14 @@ def scale_noise(
         if self.step_index is None:
             self._init_step_index(timestep)
 
-        sigma = self.sigmas[self.step_index]
+        sigmas = self.sigmas.to(device=sample.device, dtype=sample.dtype)
```

Comment:
currently, we rely on the `step_index` counter to find sigmas; 
with the new code, we will use search to find the index - we moved away from this behavior for a couple of reasons: 
* bugs caused by duplicated timesteps ( I don't think this is relevant to flow match yet, there won't be duplicated timesteps for now)
* it is also a surprisingly pretty expensive operation https://github.com/huggingface/diffusers/issues/3950) 

is the `timesteps` passed here an array of different values?  is the`begin_index` `None` in this case? can you explain a little bit more how the scale_noise is used for diff-diff and maybe you can take a look at https://github.com/huggingface/diffusers/blob/589931ca791deb8f896ee291ee481070755faa26/src/diffusers/schedulers/scheduling_euler_discrete.py#L618 and see how can we handle the diff-diff differently while keep the default behavior unchanged

Reply:
I see, we can do different approach then, didn't know it made the generation that slower, I didn't notice any kind of slowdown.

yeah,  `timesteps" is an array of different values, first I checked that if had more than one item do this, but tried to keep it more simple, also the previous error in the image to image pippeline was interfering because this array was an array with the same size as the inference steps.

Basically what we need for diff-diff is this:

```python
timesteps, num_inference_steps = self.get_timesteps(num_inference_steps, strength, device)

original_with_noise = self.prepare_latents(
            original_image, timesteps, batch_size, num_images_per_prompt, prompt_embeds.dtype, device, generator
        )
```

and then in the denoising loop:

```python
                if i == 0:
                    latents = original_with_noise[:1]
                else:
                    ...
                    latents = original_with_noise[i] * mask + latents * (1 - mask)
```

So what we need for diff-diff is to be able to pass all the timesteps and that it returns the noise latents for each one.

So I think we have 3 choices here:

- do a check in `scale_noise` and do this just in the case of when `timesteps` has more than one. Also `begin_index` is `None` 
- add a new `add_noise` function which is what diff-diff originally used.
- Just patch the scheduler inside the community pipeline and make it a self contained solution only.

I'm fine doing any of them.

Reply:
I left a comment here https://github.com/huggingface/diffusers/pull/8679/files#r1652134175